### ISSUE TITLE
Change might be needed for TemplateHaskell shipped with GHC 9.8

### ISSUE THREAD

**Issue by andreasabel** [stevenfontanella/microlens#170]
Here is a patch that might become relevant after GHC 9.8 is released (don't know yet, but wanted to drop the pointer here):
https://gitlab.haskell.org/ghc/head.hackage/-/merge_requests/302/diffs#diff-content-ded2c3d3ecf900ee20b6404ad7c2d6dbd2093f70

**Comment by andreasabel** [comment]
With GHC 9.8.1 alpha1 available, the build error is now manifest:
```
src/Lens/Micro/TH.hs:805:41: error: [GHC-83865]
    • Couldn't match type ‘()’ with ‘BndrVis’
      Expected: [TyVarBndr BndrVis]
        Actual: [TyVarBndrUnit]
    • In the second argument of ‘(:)’, namely ‘vars’
      In the third argument of ‘classD’, namely ‘(plainTV c : vars)’
      In the first argument of ‘($)’, namely
        ‘classD (cxt []) className (plainTV c : vars) fd’
    |
805 |   classD (cxt[]) className (D.plainTV c:vars) fd
    |                                      
```

**Comment by stevenfontanella** [comment]
Released [microlens-th-0.4.3.14](https://hackage.haskell.org/package/microlens-th-0.4.3.14). We can follow up in #171 on the `text` version bounds.
